### ISSUE TITLE
[Question] Connect-DbaInstance does not return suitable values in some properties - is this intended?

### ISSUE THREAD

**Issue by andreasjordan** [sqlcollaborative/dbatools#6856]
I did some testing and would be happy if others can run similar test in their environments.

My code:

```
$credentialSa = Get-Credential

[DbaInstance]$instanceV1 = 'sqlix.ordix.de\pstest,14331'        # this format will be used in .FullSmoName
[DbaInstance]$instanceV2 = 'sqlix.ordix.de\pstest:14331'        # 
[DbaInstance]$instanceV3 = 'sqlix.ordix.de,14331\pstest'        # 
[DbaInstance]$instanceV4 = 'sqlix.ordix.de:14331\pstest'        # this format will be used in .FullName
[DbaInstance]$instanceV5 = 'testsql.domain.local\pstest,14331'  # same server, but with alias
[DbaInstance]$instanceV6 = '192.168.6.29\pstest,14331'          # same server, but witt ipaddress

$serverV1 = Connect-DbaInstance -SqlInstance $instanceV1 -SqlCredential $credentialSa
$serverV2 = Connect-DbaInstance -SqlInstance $instanceV2 -SqlCredential $credentialSa
$serverV3 = Connect-DbaInstance -SqlInstance $instanceV3 -SqlCredential $credentialSa
$serverV4 = Connect-DbaInstance -SqlInstance $instanceV4 -SqlCredential $credentialSa
$serverV5 = Connect-DbaInstance -SqlInstance $instanceV5 -SqlCredential $credentialSa
$serverV6 = Connect-DbaInstance -SqlInstance $instanceV6 -SqlCredential $credentialSa

[DbaInstance]$serverV1asInstance = $serverV1
[DbaInstance]$serverV2asInstance = $serverV2
[DbaInstance]$serverV3asInstance = $serverV3
[DbaInstance]$serverV4asInstance = $serverV4
[DbaInstance]$serverV5asInstance = $serverV5
[DbaInstance]$serverV6asInstance = $serverV6

[DbaInstance]$serverV1asInstanceFromName = $serverV1.Name
[DbaInstance]$serverV2asInstanceFromName = $serverV2.Name
[DbaInstance]$serverV3asInstanceFromName = $serverV3.Name
[DbaInstance]$serverV4asInstanceFromName = $serverV4.Name
[DbaInstance]$serverV5asInstanceFromName = $serverV5.Name
[DbaInstance]$serverV6asInstanceFromName = $serverV6.Name

$instanceV1, $instanceV2, $instanceV3, $instanceV4, $instanceV5, $instanceV6 | 
    Format-Table -Property InputObject, ComputerName, InstanceName, Port, FullName, FullSmoName, SqlComputerName, SqlInstanceName, SqlFullName

$serverV1, $serverV2, $serverV3, $serverV4, $serverV5, $serverV6 | 
    Format-Table -Property Name, ComputerName, DbaInstanceName, NetPort, NetName, DomainInstanceName, ComputerNamePhysicalNetBIOS

$serverV1asInstance, $serverV2asInstance, $serverV3asInstance, $serverV4asInstance, $serverV5asInstance, $serverV6asInstance | 
    Format-Table -Property InputObject, ComputerName, InstanceName, Port, FullName, FullSmoName, SqlComputerName, SqlInstanceName, SqlFullName

$serverV1asInstanceFromName, $serverV2asInstanceFromName, $serverV3asInstanceFromName, $serverV4asInstanceFromName, $serverV5asInstanceFromName, $serverV6asInstanceFromName | 
    Format-Table -Property InputObject, ComputerName, InstanceName, Port, FullName, FullSmoName, SqlComputerName, SqlInstanceName, SqlFullName
```

The output:

![image](https://user-images.githubusercontent.com/66946165/93489316-a8c82e00-f907-11ea-89c5-c4e8b8a53e5c.png)

The first output shows how the string is parsed to get the different parts and that there are different ways to arrange the different parts.

The second output shows the resulting smo object after connecting to the sql server.
"Name" is the FullSmoName of the input object and is the only property with the detailed information about how to get to the sql server.
"ComputerName" is an added property and derived from NetName (property of the official smo) and now missing the domain part.
"DbaInstanceName" is an added property and is the InstanceName of the input object.
"NetPort" is an added property and is the Port of the input object.
"NetName" is part of the official smo and gives the name of the machine the instance is running of (has to be tested on a failover cluster instance - probably gives server property machinename).
"DomainInstanceName" is part of the official smo (it has the instance name in uppercase).
"ComputerNamePhysicalNetBIOS" is part of the official smo and is the physical machine the instance is currently running on.

The third output shows the resulting smo object when converted to an object of type "DbaInstance" - like when the server smo is passed to a command as SqlInstance parameter.
Compared to the original object, all the details are only present in the property InputObject, but all the others are missing the domain info or the ipaddress.
So using ComputerName to connect to the server might not be possible.

The forth output shows the resulting smo object when converted to an object of type "DbaInstance", but this time the Name property of the smo is used.
Because this is the property where the original FullSmoName is held, this object has the same detailed information like the original one.

I would expect to have the "name of the computer" I have to use to connect to it (a simple name, a full qualified name, an alias or an ip address) to always be present in the property ComputerName like I have it in the first output.

**Comment by wsmelton** [comment]
Please update the report using our template so it clearly stated what the expected and actual behavior is and what has to be addressed.

**Comment by andreasjordan** [comment]
Actual behavior is shown in code and screenshots, expected behavior has two sides: My personal is clearly stated at the end of the issue: 
> I would expect to have the "name of the computer" I have to use to connect to it (a simple name, a full qualified name, an alias or an ip address) to always be present in the property ComputerName like I have it in the first output.

The expected behavior of the authors of dbatools might be just the actual behavior. But that's basically what I want to know.

**Comment by wsmelton** [comment]
Please use our template.

**Comment by andreasjordan** [comment]
Solved with the experimental code path from #6904.
